BookPlayer/Models/Book+CoreDataClass.swift

```diff
@@ -106,10 +106,45 @@ public class Book: LibraryItem {
 
         let legacyIdentifier = bookUrl.originalUrl.lastPathComponent
         let storedTime = UserDefaults.standard.double(forKey: legacyIdentifier)
-        //migration of time
+
+        // migration of time
         if storedTime > 0 {
             self.currentTime = storedTime
             UserDefaults.standard.removeObject(forKey: legacyIdentifier)
         }
     }
+
+    override func getBookToPlay() -> Book? {
+        return self
+    }
+
+    func nextBook() -> Book? {
+        if
+            let playlist = self.playlist,
+            let next = playlist.getNextBook(after: self) {
+            return next
+        }
+
+        guard
+            UserDefaults.standard.bool(forKey: Constants.UserDefaults.autoplayEnabled.rawValue),
+            let library = self.library
+        else {
+            return nil
+        }
+
+        let nextItem = library.getNextItem(after: self.playlist ?? self)
```

Comment:
also crashing when library autoplay is enabled and the last item finishes playing